script/fetchneedles

```diff
@@ -51,7 +52,9 @@ git_update() {
     branch="${1:-"master"}"
     git gc --auto --quiet
     git fetch -q origin
-    git rebase -q origin/"$branch"
+    # Clear any uncommitted changes that would prevent a rebase
+    [ "$force" = 1 ] && git reset --hard HEAD
+    git rebase -q origin/"$branch" || echo "Use force=1 to discard uncommitted changes before rebasing" && exit 1
```

Comment:
this won't work either, try

```bash
echo "pass" || echo "fail" && echo "exit 1"
```

locally to see what I mean. You will see

```
pass
exit 1
```

See https://github.com/os-autoinst/openQA/blob/master/script/openqa-clone-custom-git-refspec#L36 for a solution.

Define

```bash
fail() {
    echo "$*" >&2
    exit 1
}
```

then call

```suggestion
    git rebase -q origin/"$branch" || fail "Use force=1 to discard uncommitted changes before rebasing"
```

Reply:
I'm confused. I tested it and it worked perfectly. I guess it might depend on the shell or whatever 😒

Will use the *fail* function. Thank you for the idea.